### DISCUSSION TITLE
Cannot Ping Server once VPN Tunnel created

### DISCUSSION THREAD

**Discussion by Bluemesh-Solutions** [strongswan/strongswan#2303] [Q&A]
I have two raspberry Pi's. 

The server has I.P 192.168.0.65 
**Server ipsec.conf**

config setup
	charondebug = "ike 2, knl 1"  # Enable debugging (optional)

conn %default

conn ipsec-conn
	keyexchange=ikev2
	auto=add
	left=192.168.0.65 
	right=%any 
	esp=aes256-sha256-modp2048!
	ike=aes256-sha256-modp2048!  
    	authby = psk

The client has I.P 192.168.0.60
**Client  ipsec.conf**

config setup
        charondebug = "ike 2, knl 1"  # Enable debugging (optional)

conn %default

conn ipsec-conn
        keyexchange=ikev2
        auto=add
        left=192.168.0.65
        right=192.168.0.60
	    esp=aes256-sha256-modp2048!
        ike=aes256-sha256-modp2048!  instead of certificates
        authby = psk

Server ipsec.secrets
192.168.0.65 192.168.0.60 : PSK "Jrw92hCusu9SVPMKgqsN1p5QdiPrrdq-BCHVAsXxym4="

Client ipsec.secrets
192.168.0.60 192.168.0.65 : PSK "Jrw92hCusu9SVPMKgqsN1p5QdiPrrdq-BCHVAsXxym4="

On the client I have a VPN Tunnel Connection

**sudo cat /etc/NetworkManager/system-connections/VPN-Client**
[connection]
id=VPN-Client
uuid=94fa2fdc-5251-47d8-a0f3-bd62c0a4888a
type=vpn
autoconnect=false

[vpn]
address=192.168.0.65
encap=no
ipcomp=no
method=psk
password-flags=0
proposal=no
virtual=no
service-type=org.freedesktop.NetworkManager.strongswan

[vpn-secrets]
password=Jrw92hCusu9SVPMKgqsN1p5QdiPrrdq-BCHVAsXxym4=

[ipv4]
method=auto

[ipv6]
addr-gen-mode=stable-privacy
method=auto

[proxy] 

>**ping 192.168.0.65**
PING 192.168.0.65 (192.168.0.65) 56(84) bytes of data.
64 bytes from 192.168.0.65: icmp_seq=1 ttl=64 time=5.04 ms
64 bytes from 192.168.0.65: icmp_seq=2 ttl=64 time=7.67 ms
64 bytes from 192.168.0.65: icmp_seq=3 ttl=64 time=7.96 ms

Now I start up the VPN Tunnel on the client
>**sudo nmcli con up id VPN-Client** 
Connection successfully activated (D-Bus active path: /org/freedesktop/NetworkManager/ActiveConnection/3)

On the server 
>**sudo ipsec status**
Security Associations (1 up, 0 connecting):
  ipsec-conn[2]: ESTABLISHED 60 seconds ago, 192.168.0.65[192.168.0.65]...192.168.0.60[192.168.0.60]
  ipsec-conn{1}:  INSTALLED, TUNNEL, reqid 1, ESP SPIs: c2f4771d_i c16e62f7_o
  ipsec-conn{1}:   192.168.0.65/32 === 192.168.0.60/32

On the client 
>**sudo ipsec status**
Security Associations (0 up, 0 connecting):
  none

>ping 192.168.0.65
PING 192.168.0.65 (192.168.0.65) 56(84) bytes of data.
Just hangs

Can anyone see anything obvious that I'm not doing.

**Comment by tobiasbrunner** [comment]
You seem to mix a configs via ipsec.conf and NetworkManager. The former uses the charon daemon, while the latter uses charon-nm and is not configured or controlled via ipsec.conf/ipsec. So you probably want to use one or the other. And for such a simple host-to-host connection I'd recommend using ipsec.conf/ipsec (the NetworkManager plugin is intended for roadwarriors) and you could probably set `auto=route` so the connection is triggered by matching traffic.

**Reply by Bluemesh-Solutions** [reply]
Thanks Tobias,
The **/etc/NetworkManager/system-connections/VPN-Client** file was created by selecting Wifi->Advanced Options->Add VPN Connection on the Client Pi.
I then used **sudo nmcli con up id VPN-Client** to start the VPN tunnel.

If I don't use this VPN-Client file how can I initialise the VPN Tunnel connection?

Should I be using one of the below even though neither seem to be installed with the strongswan package.

**swanctl --name <connection_name> connect**  
or
**sudo charon-cmd --host <server_hostname_or_ip> --identity <client_identity> [options]**

**Reply by tobiasbrunner** [reply]
You seem quite confused. Maybe start with the [introduction](https://docs.strongswan.org/docs/5.9/howtos/introduction.html)? And the swanctl tool is in a separate package (`strongswan-swanctl`, but it's only required to use swanctl/vici, not when using the legacy ipsec.conf/ipsec). charon-cmd is a standalone command line client, it is probably not what you want for this kind of setup.
